src/js/game/blueprint.js

```diff
@@ -57,6 +57,37 @@ export class Blueprint {
         return new Blueprint(newEntities);
     }
 
+    /**
+     * Creates a new blueprint from the given entity uids
+     * @param {Array<Entity>} entities
+     */
+    static fromEntities(entities) {
+        const newEntities = [];
```

Comment:
Let it know at once you want it to hold the same number of elements.

```suggestion
        const newEntities = [];
        newEntities.length = entities.length;
```

Or, use the Array constructor to do the same thing.
```suggestion
        const newEntities = new Array(entities.length);
```